MantidQt/MantidWidgets/src/DataProcessorUI/DataProcessorGenerateNotebook.cpp

```diff
@@ -258,24 +259,35 @@ std::string tableString(QDataProcessorTableModel_sptr model,
   table_string << "---"
                << "\n";
 
-  for (auto rowIt = rows.begin(); rowIt != rows.end(); ++rowIt) {
+  for (auto groupIt = rows.begin(); groupIt != rows.end(); ++groupIt) {
 
-    for (int col = 0; col < ncols - 1; col++)
-      table_string
-          << model->data(model->index(*rowIt, col)).toString().toStdString()
-          << " | ";
+    auto groupId = groupIt->first;
+    auto rowSet = groupIt->second;
 
-    table_string
-        << model->data(model->index(*rowIt, ncols - 1)).toString().toStdString()
-        << "\n";
-  }
+    for (auto rowIt = rowSet.begin(); rowIt != rowSet.end(); ++rowIt) {
```

Comment:
Not sure if the interface of the Qt index() methods will allow it, but this could also be a `const auto &` range-based loop? Or use `cbegin()`, `cend()`?